drivers/lsm6ds3/lsm6ds3.c

```diff
@@ -0,0 +1,296 @@
+/*
+ * Copyright (C) 2014 Freie Universität Berlin
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ */
+
+/**
+ * @ingroup     driver_lsm6ds3
+ * @{
+ *
+ * @file
+ * @brief       Device driver implementation for the LSM6DS3 3D accelerometer
+ *              and 3D gyroscope
+ *
+ * @author      René Herthel <rene-herthel@outlook.de>
+ *
+ * @}
+ */
+
+#include "lsm6ds3.h"
+#include "include/lsm6ds3-internal.h"
+
+#define ENABLE_DEBUG (1)
+#include "debug.h"
+
+int lsm6ds3_init(lsm6ds3_t *dev,
+                 i2c i2c,
+                 gpio_t acc_pin,
+                 gpio_t gyr_pin,
+                 uint8_t acc_address,
+                 lsm6ds3_acc_sample_rate_t sample_rate,
+                 lsm6ds3_acc_scale_t acc_scale,
+                 lsm6ds3_acc_bandwith_t acc_bandwith,
+                 uint_8_t gyr_address,
+                 lsm6ds3_gyr_sample_rate_t gyr_sample_rate,
+                 lsm6ds3_gyr_scale_t gyr_scale,
+                 lsm6ds3_gyr_scale_125_t gyr_scale_125)
+{
+    int res = 0;
+    char tmp;
+
+	  dev->i2c = i2c;
+	  dev->acc_address = acc_address;
+	  dev->gyr_address = gyr_address;
+	  dev->acc_pin = acc_pin;
+	  dev->gyr_pin = gyr_pin;
+
+	  i2c_acquire(dev->i2c);
+	  i2c_init_master(i2c, I2C_SPEED_NORMAL);
+	  DEBUG("LSM6DS3: reboot ... ");
+    res += i2c_write_reg(dev->i2c, dev->acc_address,
```

Comment:
As this is the first value for `res` in this function you could have just used a `=`. Anyway, here it is not that dramatically, but some lines below I think you did a mistake, will comment there. Still I think you don't need the `if(res != ...)` after each i2c transfer. The last one should be fine.